### PR TITLE
FunctionalTest always use Java 8 to run the build & set testJavaHome

### PR DESCRIPTION
**Pull request by eskatos**
to run tests using the selected JVM.

Follow up for #3735

### CODE REVIEW

**Comment by bamboo** [comment]
\o/

File: .teamcity/Gradle_Check/configurations/FunctionalTest.kt [hunk 1]
```diff
@@ -16,10 +16,11 @@ class FunctionalTest(model: CIBuildModel, testCoverage: TestCoverage, subProject
     } + testCoverage.testType.name + "Test"
     val quickTest = testCoverage.testType == TestType.quick
     applyDefaults(model, this, testTask, notQuick = !quickTest, os = testCoverage.os,
+            extraParameters = "-PtestJavaHome=%${testCoverage.os}.${testCoverage.version}.${testCoverage.vendor}.64bit%",
```
Comment (wolfs) [inline .teamcity/Gradle_Check/configurations/FunctionalTest.kt]: ❌ Please quote the property, e.g. like we do here: https://github.com/gradle/gradle/blob/240521ec79cf094a15f0ae7c3f99ce2702e1801a/.teamcity/Gradle_Check/configurations/GradleBuildConfigurationDefaults.kt#L14

Comment (eskatos) [reply .teamcity/Gradle_Check/configurations/FunctionalTest.kt]: 👌 ✅

**Comment by eskatos** [comment]
@wolfs good points, let's restrict this PR to `FunctionalTest` and move forward

**Pull request merged**